ssi-ldp/src/suites/rss.rs

```diff
@@ -0,0 +1,200 @@
+        key: &JWK,
+        extra_proof_properties: Option<HashMap<String, Value>>,
+    ) -> Result<Proof, Error> {
+        // TODO: add proof context?
+        let mut proof = Proof::new(ProofSuiteType::RSSSignature2023)
+            .with_options(options)
+            .with_properties(extra_proof_properties);
+        let quads = doc_to_sorted_quads(document, context_loader).await?;
+        let msgs = quads
+            .into_iter()
+            .map(|q| FieldElement::from_msg_hash(q.as_bytes()))
+            .collect::<Vec<_>>();
+        let rss_keys: RSSKeyPair = key
+            .try_into()
+            .map_err(|err: RSSKeyError| ssi_jwk::Error::from(err))?;
+
+        let sig = RSignature::new(&msgs, &rss_keys.private_key);
+        proof.proof_value = Some(sig.to_hex());
+        Ok(proof)
+    }
+
+    pub(crate) async fn verify(
+        proof: &Proof,
+        document: &(dyn LinkedDataDocument + Sync),
+        resolver: &dyn DIDResolver,
+        context_loader: &mut ContextLoader,
+    ) -> Result<VerificationWarnings, Error> {
+        let sig_hex = proof
+            .proof_value
+            .as_ref()
+            .ok_or(Error::MissingProofSignature)?;
+        let verification_method = proof
+            .verification_method
+            .as_ref()
+            .ok_or(Error::MissingVerificationMethod)?;
+        let vm = resolve_vm(verification_method, resolver).await?;
+        // TODO: update with rss type once appropriate context is available
+        if vm.type_ != "JsonWebSignature2020" {
+            todo!("checking vm type, found: {}", vm.type_);
+            return Err(Error::VerificationMethodMismatch);
+        }
+        let jwk = vm.public_key_jwk.ok_or(Error::MissingKey)?;
+
+        let InferredDataset {
+            null_marked_dataset_quads,
+            inferred_idxs,
+        } = infer_disclosed_idxs(document, context_loader).await?;
+
+        let msgs = null_marked_dataset_quads
+            .into_iter()
+            .enumerate()
+            .map(|(i, q)| {
+                if inferred_idxs.contains(&(i + 1)) {
+                    FieldElement::from_msg_hash(q.as_bytes())
+                } else {
+                    FieldElement::zero()
+                }
+            })
+            .collect::<Vec<_>>();
+
+        let res = RSignature::verifyrsignature(
+            &jwk.try_into()
+                .map_err(|e: RSSKeyError| <RSSKeyError as Into<ssi_jwk::error::Error>>::into(e))?,
+            &RSignature::from_hex(sig_hex)
+                .map_err(|e| <RSignatureError as Into<RSSError>>::into(e))?,
+            &msgs,
+            &inferred_idxs,
+        );
+
+        match res {
+            RSVerifyResult::Valid => Ok(vec![]),
+            err @ _ => Err(<RSVerifyResult as Into<RSSError>>::into(err).into()),
+        }
+    }
+}
+
+pub struct InferredDataset {
+    pub inferred_idxs: Vec<usize>,
+    pub null_marked_dataset_quads: Vec<String>,
+}
+
+const NULL_MARKER: &str = "__12345__";
```

Comment:
Is there a test to prove that the algorithm is insensitive to the choice of null marker/ whether or not the null marker appears as a value in the credential?